src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2492 @@
+                A tensor of shape `batch_size, num_classes`
+            auxilary_predictions (`Dict[str, torch.Tensor]`, *optional*):
+                if `use_auxilary_loss` was set to `true` in [`MaskFormerConfig`], then it contains the logits from the
+                inner layers of the Detr's Decoder.
+
+        Returns:
+            `Dict[str, Tensor]`: A dict of `torch.Tensor` containing two keys:
+            - **loss_cross_entropy** -- The loss computed using cross entropy on the predicted and ground truth labels.
+            - **loss_mask** -- The loss computed using sigmoid focal loss on the predicted and ground truth masks.
+            - **loss_dice** -- The loss computed using dice loss on the predicted on the predicted and ground truth
+              masks.
+            if `use_auxilary_loss` was set to `true` in [`MaskFormerConfig`], the dictionary contains addional losses
+            for each auxilary predictions.
+        """
+
+        # Retrieve the matching between the outputs of the last layer and the labels
+        indices = self.matcher(masks_queries_logits, class_queries_logits, mask_labels, class_labels)
+
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks: Number = self.get_num_masks(class_labels, device=class_labels.device)
+
+        # Compute all the requested losses
+        losses: Dict[str, Tensor] = {
+            **self.loss_masks(masks_queries_logits, mask_labels, indices, num_masks),
+            **self.loss_labels(class_queries_logits, class_labels, indices),
+        }
+
+        # In case of auxiliary losses, we repeat this process with the output of each intermediate layer.
+        if auxilary_predictions is not None:
+            for idx, aux_outputs in enumerate(auxilary_predictions):
+                masks_queries_logits = aux_outputs["masks_queries_logits"]
+                class_queries_logits = aux_outputs["class_queries_logits"]
+                loss_dict = self.forward(masks_queries_logits, class_queries_logits, mask_labels, class_labels)
+                loss_dict = {f"{key}_{idx}": value for key, value in loss_dict.items()}
+                losses.update(loss_dict)
+
+        return losses
+
+    def get_num_masks(self, class_labels: torch.Tensor, device: torch.device) -> torch.Tensor:
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks = class_labels.shape[0]
+        num_masks_pt = torch.as_tensor([num_masks], dtype=torch.float, device=device)
+        return num_masks_pt
+
+
+class MaskFormerSwinTransformerBackbone(nn.Module):
+    """
+    This class uses [`MaskFormerSwinModel`] to reshape its `hidden_states` from (`batch_size, sequence_length,
+    hidden_size)` to (`batch_size, num_channels, height, width)`).
+
+    Args:
+        config (`SwinConfig`):
+            The configuration used by [`MaskFormerSwinModel`].
+    """
```

Comment:
Wondering if we can't make this a bit simpler, by:
* making MaskFormerSwinModel directly return the reshaped hidden states of shape (B, C, H, W).

=> in that case, the forward of this class becomes a lot easier.

Reply:
I'd argue that won't make it simpler but just move complexity to another part. I can return the already resized features from Swin but it requires a lot of refactoring and I am not sure if it is easier to follow. What do you think?